### PR TITLE
ACS quickstart for JobRouter with AzureOpenAI

### PR DESCRIPTION
**Pull request by NathanJBennett**
## Purpose
QuickStart guide to add Azure OpenAI into your Job Router ACS resource for matching.

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?
Quick start application

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Clone repo setup resources as noted in Readme. Run application, check logs in Azure function to monitor output of OpenAI

## What to Check
Verify that the following are valid
* Check scores given to workers viewing the Azure Function logs.

### CODE REVIEW

**Review by adamtuck-msft — APPROVED** [review]

File: JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+                if (userInput == "yes")
+                {
+                    var jobId = await CreateJob();
+
+                    var waitForOffer = true;
+
+                    while (waitForOffer)
+                    {
+                        Task.Delay(new TimeSpan(0,0,2));
```
Comment (williamzhao87) [inline JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs:44]:
```suggestion
                        Task.Delay(TimSpan.FromMinutes(2));
```

File: JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+                        Console.WriteLine("Checking for Offers");
+
+                        await foreach (var worker in JobRouterClient.GetWorkersAsync(queueId: Configuration["JobRouterResources:Queue:id"]))
+                        { 
+                            if (worker.Offers.Count > 0 && worker.Offers.FirstOrDefault().JobId == jobId)
+                            {
+                                Console.WriteLine($"Worker {worker.Id} has received the offer for new Job. Taking the Job to Closes state.");
+                                var acceptOffer = await JobRouterClient.AcceptJobOfferAsync(worker.Id, worker.Offers.FirstOrDefault().OfferId);
+                                Task.Delay(new TimeSpan (0,0,2));
```
Comment (williamzhao87) [inline JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs:54]:
```suggestion
                                Task.Delay(TimSpan.FromMinutes(2));
```

File: JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+                                await JobRouterClient.CompleteJobAsync(new CompleteJobOptions(jobId, acceptOffer.Value.AssignmentId));
+                                Task.Delay(new TimeSpan(0, 0, 1));
```
Comment (williamzhao87) [inline JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs:56]:
```suggestion
                                Task.Delay(TimSpan.FromMinutes(1));
```

File: JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            return jobOptions.JobId;
+        }
+
+        private static BestWorkerMode SetBestWorkerOpenAiMode()
+        {
+            var mode = new BestWorkerMode() { ScoringRule = new FunctionRouterRule(new Uri(Configuration["OpenAiPairing:AzureFunctionUri"]))
+            {
+                Credential = new FunctionRouterRuleCredential(Configuration["OpenAiPairing:AzureFunctionKey"])
+            },
+                MinConcurrentOffers = 1,
```
Comment (williamzhao87) [inline JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs:211]: Min/Max default to 1, so they do not need to be set

**Review by williamzhao87 — APPROVED** [review]

File: JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs (hunk 1 above)
Comment (NathanJBennett) [reply JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs]: Updated

File: JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs (hunk 2 above)
Comment (NathanJBennett) [reply JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs]: Updated

File: JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs (hunk 4 above)
Comment (NathanJBennett) [reply JobRouterOpenAIIntegration/JR_AOAI_Integration/Program.cs]: Removed

**Pull request merged**